api/tests/unit/domain/services/mail-service_test.js

```diff
@@ -19,56 +24,115 @@ describe('Unit | Service | MailService', () => {
 
   describe('#sendAccountCreationEmail', () => {
 
-    it('should use mailer to send an email with locale "fr-fr"', async () => {
-      // given
+    it('should call sendEmail with from, to, subject, template', async () => {
+
       const locale = 'fr-fr';
-      const domainFr = 'pix.fr';
+
+      // given
       const expectedOptions = {
         from: senderEmailAddress,
-        fromName: 'PIX - Ne pas répondre',
         to: userEmailAddress,
         subject: 'Votre compte Pix a bien été créé',
         template: 'test-account-creation-template-id',
-        variables: {
-          homeName: `${domainFr}`,
-          homeUrl: `https://${domainFr}`,
-          redirectionUrl: `https://app.${domainFr}/connexion`,
-          locale,
-        },
       };
 
       // when
       await mailService.sendAccountCreationEmail(userEmailAddress, locale);
 
       // then
-      expect(mailer.sendEmail).to.have.been.calledWithExactly(expectedOptions);
+      const options = mailer.sendEmail.firstCall.args[0];
+      expect(options).to.include(expectedOptions);
     });
 
-    it('should use mailer to send an email with redirectionUrl from parameters', async () => {
-      // given
-      const redirectionUrl = 'https://pix.fr';
-      const locale = 'fr-fr';
-      const domainFr = 'pix.fr';
-      const expectedOptions = {
-        from: senderEmailAddress,
-        fromName: 'PIX - Ne pas répondre',
-        to: userEmailAddress,
-        subject: 'Votre compte Pix a bien été créé',
-        template: 'test-account-creation-template-id',
-        variables: {
-          homeName: `${domainFr}`,
-          homeUrl: `https://${domainFr}`,
-          redirectionUrl,
-          locale,
-        },
-      };
+    context('according to redirectionUrl', () => {
 
-      // when
-      await mailService.sendAccountCreationEmail(userEmailAddress, locale, redirectionUrl);
+      context('if redirectionUrl is provided', () => {
```

Comment:
Et si `redirectionUrl` n'est pas fourni ?

Reply:
yep, je vais faire comme https://github.com/1024pix/pix/pull/2655#discussion_r588169287

Reply:
Suggestion implémentée

Reply:
Dans ce cas, il est positionné en fonction de la locale et testé dans le contexte `according to locale`